### PR TITLE
Added code for sqrtdecomposition (MO's Algorithm)in python

### PR DESCRIPTION
**Pull request by abhijeet010304**
**Fixes issue:**
 Created code for MO's algorithm in python3.

**Changes:**

Added code for MO's algorithm in python3.

### CODE REVIEW

File: code/square_root_decomposition/src/mos_algorithm/sqrtdecomposition.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+
+buckets = int(math.ceil(math.sqrt(n)))
+total_odds = [0 for i in range(buckets)]
+
+bucket_size = int(math.ceil(n / buckets))
+
+
+
+for bucket in range(buckets):
```
Comment (arnavb) [inline code/square_root_decomposition/src/mos_algorithm/sqrtdecomposition.py]: Please put all of the algorithm code itself in its own function.

File: code/square_root_decomposition/src/mos_algorithm/sqrtdecomposition.py [hunk 2]
```diff
@@ -0,0 +1,55 @@
+# PROBLEM STATEMENT-> You are given an array A of size N and Q queries.(All indexes are 1 based).
+# Each query can be of two types
+#       1->   1 i  - add 1 to the ith element of the array A
+#       2->   2 K  - print the Kth odd number of the array A if it exists, else print -1
+
+import math
+n, q = list(map(int, input().strip().split(' ')))
```
Comment (arnavb) [inline code/square_root_decomposition/src/mos_algorithm/sqrtdecomposition.py]: This should all go under an `if __name__ == '__main__':`.

File: code/square_root_decomposition/src/mos_algorithm/sqrtdecomposition.py [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+            total_odds[bucket] += 1
+        else:
+            total_odds[bucket] +=-1
+    else:
+        cnt = 0
+        found_at = -1
+        for index, count in enumerate(total_odds):
+            cnt += count
+            if cnt >= k:
+
```
Comment (arnavb) [inline code/square_root_decomposition/src/mos_algorithm/sqrtdecomposition.py]:
```suggestion
```

File: code/square_root_decomposition/src/mos_algorithm/sqrtdecomposition.py [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+                start = bucket * bucket_size
+                end = min(n, start + bucket_size)
+
+                cnt -= count
+                for j in range(start, end):
+                    if array[j] %2 != 0:
+                        cnt += 1
+                        if cnt == k:
+                            found_at = j + 1
+
```
Comment (arnavb) [inline code/square_root_decomposition/src/mos_algorithm/sqrtdecomposition.py]:
```suggestion
```

File: code/square_root_decomposition/src/mos_algorithm/sqrtdecomposition.py (hunk 1 above)
Comment (abhijeet010304) [reply code/square_root_decomposition/src/mos_algorithm/sqrtdecomposition.py]: OK sir .

File: code/square_root_decomposition/src/mos_algorithm/sqrtdecomposition.py (hunk 2 above)
Comment (abhijeet010304) [reply code/square_root_decomposition/src/mos_algorithm/sqrtdecomposition.py]: OK sir got it.

File: code/square_root_decomposition/src/mos_algorithm/sqrtdecomposition.py (hunk 1 above)
Comment (abhijeet010304) [reply code/square_root_decomposition/src/mos_algorithm/sqrtdecomposition.py]: done sir! please review my latest PR
 https://github.com/OpenGenus/cosmos/pull/4663

File: code/square_root_decomposition/src/mos_algorithm/sqrtdecomposition.py (hunk 2 above)
Comment (abhijeet010304) [reply code/square_root_decomposition/src/mos_algorithm/sqrtdecomposition.py]: Done Sir! please review my latest PR  https://github.com/OpenGenus/cosmos/pull/4663

File: code/square_root_decomposition/src/mos_algorithm/sqrtdecomposition.py [hunk 5]
```diff
@@ -0,0 +1,66 @@
+# PROBLEM STATEMENT-> You are given an array A of size N and Q queries.(All indexes are 1 based).
```
Comment (arnavb) [inline code/square_root_decomposition/src/mos_algorithm/sqrtdecomposition.py]: Reformat this file using black.

Comment (abhijeet010304) [reply code/square_root_decomposition/src/mos_algorithm/sqrtdecomposition.py]: Sir in my last PR , Firstly I have copied my code pasted in the left side of the black then copied from the right side and saved into a file . Then I pushed that file in gituhub. 
Is there any other procedure to reformat it??

Comment (arnavb) [reply code/square_root_decomposition/src/mos_algorithm/sqrtdecomposition.py]: No you are correct. My mistake I was looking at the wrong file in the Travis output.

**Comment by arnavb** [comment]
Nice job and welcome to cosmos!

File: code/square_root_decomposition/src/mos_algorithm/sqrtdecomposition.py (hunk 5 above)
Comment (abhijeet010304) [reply code/square_root_decomposition/src/mos_algorithm/sqrtdecomposition.py]: Thanks a lot for merging ...

**Pull request merged**